### PR TITLE
[4.0] Add missing XML files for Admin Menu creation

### PR DESCRIPTION
**Pull request by drmenzelit**
### Summary of Changes
Added several xml files for admin components to be able to create new menu items in admin menus

### CODE REVIEW

**Comment by wilsonge** [comment]
Do the fields views not need to have a component specified?

**Comment by drmenzelit** [comment]
In my last commit I added a parameter on the files for fields to choose the correct component. The new file administrator/components/com_menus/Field/ComponentsfieldsField.php does the work (at least on my tests) :-)

**Comment by chmst** [comment]
Selecting the component which use fields is not clear. There is no flag for "this component uses fields".
So the field selects components with param "custom_fields_enable":"1" . 

After a fresh installation, only com_contact has set this parameter so other components, also com_content or com_users are not in the list.

We could change the install files, so that at least all core components have this parameter. But for 3rd party components?

File: administrator/components/com_fields/tmpl/fields/default.xml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+		</message>
+	</layout>
+	<fieldset name="request">
+		<fields name="request">
+			<field
+				name="extension"
+				type="componentsfields"
+				label="COM_FIELDS_CHOOSE_CONTEXT_LABEL"
+				required="true"
+			>
```
Comment (Quy) [inline administrator/components/com_fields/tmpl/fields/default.xml]:
```suggestion
				>
```

File: administrator/components/com_fields/tmpl/groups/default.xml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+		</message>
+	</layout>
+	<fieldset name="request">
+		<fields name="request">
+			<field
+				name="extension"
+				type="componentsfieldgroup"
+				label="COM_FIELDS_CHOOSE_CONTEXT_LABEL"
+				required="true"
+			>
```
Comment (Quy) [inline administrator/components/com_fields/tmpl/groups/default.xml]:
```suggestion
				>
```

File: administrator/components/com_workflow/tmpl/workflows/default.xml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+		</message>
+	</layout>
+	<fieldset name="request">
+		<fields name="request">
+			<field
+				name="extension"
+				type="componentsworkflow"
+				label="COM_WORKFLOW_CHOOSE_CONTEXT_LABEL"
+				required="true"
+			>
```
Comment (Quy) [inline administrator/components/com_workflow/tmpl/workflows/default.xml]:
```suggestion
				>
```

**Comment by brianteeman** [comment]
Can you look at resolving the conflicts and addressing the suggestions of @quy so that this can be tested please.

**Comment by ghost** [comment]
Please resolve conflicting files so this PR can get tested at Worldwide Pizza, Bugs & Fun, October 19th

**Comment by brianteeman** [comment]
@franz-wohlkoenig I really hope it can be fixed and merged before then

**Comment by drmenzelit** [comment]
I have commited the suggested changes, I resolved the conflict on the language file and actualized my branch. I hope, it was correct and now the PR can be tested.

**Comment by VladikB** [comment]
I have tested this item :white_check_mark: successfully on 408a5ca65081e25bde518954f89b40706c970928<br /><br />New menu items:
- Content Security Policy 
  -> List All Content Security Policy Reports
- Messaging
  -> List all Messages
- Smart Search
  -> Smart Search Content Mapping
  -> Smart Search Filters
  -> Smart Search Indexed Content
  -> Smart Search results
  -> Smart Search Statistics
- Workwlows
  -> List All Workflows<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24284">issues.joomla.org/tracker/joomla-cms/24284</a>.</sub>

**Comment by tecpromotion** [comment]
I have tested this item :white_check_mark: successfully on 408a5ca65081e25bde518954f89b40706c970928<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24284">issues.joomla.org/tracker/joomla-cms/24284</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24284">issues.joomla.org/tracker/joomla-cms/24284</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Comment by wilsonge** [comment]
@chmst I think we should add that default value too the installation sql too

**Comment by wilsonge** [comment]
Codestyle fixed with https://github.com/joomla/joomla-cms/commit/22feb63dc1d6c9e33138caacb8fb5b4751f25ae0

**Pull request merged**